examples/resolve_names.cpp

```diff
@@ -0,0 +1,61 @@
+//   Copyright 2017 otris software AG
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+#include <ews/ews.hpp>
+#include <ews/ews_test_support.hpp>
+
+#include <cstdlib>
+#include <exception>
+#include <iostream>
+#include <ostream>
+#include <string>
+
+int main()
+{
+    int res = EXIT_SUCCESS;
+    ews::set_up();
+
+    try
+    {
+        const auto env = ews::test::environment();
+        auto service = ews::service(env.server_uri, env.domain, env.username,
+                                    env.password);
+
+        std::string name = "person";
+        std::vector<ews::folder_id> id_vector;
+        ews::folder_id folder;
+        id_vector.push_back(folder);
+        auto response = service.resolve_names(
+            name, ews::searchscope::active_directory, id_vector);
+        std::cout << response.total_items_in_view << std::endl;
+        if (response.resolutions.size() > 0)
+        {
+            auto reso = response.resolutions[0];
+            std::cout << reso.mailbox.name() << std::endl;
+            std::cout << reso.mailbox.value() << std::endl;
+            std::cout << reso.directory_id.id << std::endl;
+        }
+
```

Comment:
Is there a reason you show only the first result? You might want to make this a little more appealing by showing all results, and print the mailbox property names.